client/src/views/home/DeviceJoinOrganizationModal.vue

```diff
@@ -28,20 +28,39 @@
     <div
       class="modal"
       :class="{
-        wizardTrue: pageStep > DeviceJoinOrganizationStep.Information && pageStep != DeviceJoinOrganizationStep.Finish,
+        wizardTrue: pageStep > DeviceJoinOrganizationStep.Information && pageStep != DeviceJoinOrganizationStep.Finish && isLargeDisplay,
       }"
     >
-      <ion-header class="modal-header">
+      <ion-header
+        class="modal-header"
+        v-if="isLargeDisplay"
+      >
         <ion-title class="modal-header__title title-h2">
-          {{ $msTranslate(getTitleAndSubtitle().title) }}
+          {{ $msTranslate(steps[pageStep]?.title) }}
         </ion-title>
         <ion-text
-          v-if="getTitleAndSubtitle().subtitle"
+          v-if="steps[pageStep]?.subtitle"
           class="modal-header__text body"
         >
-          {{ $msTranslate(getTitleAndSubtitle().subtitle) }}
+          {{ $msTranslate(steps[pageStep]?.subtitle) }}
         </ion-text>
       </ion-header>
+      <template v-else>
+        <small-display-modal-header
+          v-if="pageStep === DeviceJoinOrganizationStep.Information || pageStep === DeviceJoinOrganizationStep.Finish"
+          @close-clicked="cancelModal()"
+          :hide-close-button="pageStep === DeviceJoinOrganizationStep.Finish"
+          :title="steps[pageStep]?.title"
+        />
+        <small-display-step-modal-header
+          v-else
+          @close-clicked="cancelModal()"
+          :title="'DevicesPage.addDevice'"
+          :icon="phonePortrait"
+          :steps="steps.slice(1, steps.length - 1)"
+          :current-step="pageStep"
```

Comment:
Pass the right step (here it's `pageStep - 1`). The SmallDisplayStepModalHeader component cannot know whether it should do -1 or -2 or nothing. It just uses the index you're giving it.